### DISCUSSION TITLE
用 ChatGPT 执行自动化任务：Auto-GPT

### DISCUSSION THREAD

**Discussion by adoyle-h** [just-talks/tech-talks#45] [3.分享创造]
https://github.com/Torantulino/Auto-GPT

这个项目能根据用户需求自动编排任务，并且自动化执行任务。根据 demo 可以看到，设置 1~4 个目标，然后 ChatGPT 就会自动帮用户实现功能。

它的主要特点是
- 用 ChatGPT-4 或者 ChatGPT-3.5 驱动（需要配置你自己的 OPENAI_API_KEY。记得走国外代理，不然被识别国内流量，你的 OpenAI 账号可能就被封了）
- 支持联网搜索信息，自动访问流行的网站和平台
- 支持长期和短期的会话记忆存储
- 支持文件存储
- 支持语音对话（基于 [ElevenLabs API](https://beta.elevenlabs.io/)，只支持英文。好像只有文本转语音，没有语音转文本，动手能力强的可以换成其他服务）

https://user-images.githubusercontent.com/22963551/228855501-2f5777cf-755b-4407-a643-c7299e5b6419.mp4

**Comment by timing-hbc** [comment]
稍稍研究了一下，这个产品基本上是用硬编码的方式完成了需求拆分和任务编排，微软开源的JARVIS也是采用硬编码的形式实现的，感觉路还长，论文有些过于夸大研究成果了

**Comment by keroro520** [comment]
留下两个相似的项目：

> “babyagi” is a program that given a task, creates a task list and executes the tasks over and over again. It’s now been open sourced and is the top trending repos on Github atm [[Link](https://github.com/yoheinakajima/babyagi)]. Helpful tip on running it locally [[Link](https://twitter.com/yoheinakajima/status/1643403795895058434)]. People are already working on a “toddleragi” lol [[Link](https://twitter.com/gogoliansnake/status/1643225698801164288?s=20)]

> Someone built a script that uses gpt4 to create its own code and fix its own bugs. Its basic but it can code snake by itself. Crazy potential [[Link](https://twitter.com/mattcduff/status/1642528658693984256)]
gpt自己写代码，修代码，迭代进化

**Comment by adoyle-h** [comment]
[这个视频](https://www.zhihu.com/zvideo/1626491997283397632)演示了用语音交互 GPT-4 来部署一个网站到 Vercel。
其实是 https://www.codewand.co/ 的软广。

**Comment by adoyle-h** [comment]
https://github.com/Dogtiti/AutoGPT-Next-Web

AutoGPT 的网页端工具。可以自己部署到 vercel。也可以直接用它提供的 https://auto-agentgpt.com/
